pandas/core/apply.py

```diff
@@ -131,6 +131,16 @@ def get_result(self):
 
         # ufunc
         elif isinstance(self.f, np.ufunc):
+            if any(is_sparse(dtype) for dtype in self.obj.dtypes):
+                # Column-by-column construction is slow, so only use when
+                # necessary (e.g. to preserve special dtypes) GH 23744
+                result = self.obj._constructor(index=self.index,
+                                               copy=False)
+                with np.errstate(all='ignore'):
+                    for col in self.columns:
+                        result[col] = self.f(self.obj[col].values)
+                return result
```

Comment:
as i said above, don't construct an empty Dataframe, rather use a dictionary (or just a list comprehension), *then* construct the dataframe from it right before return.

Reply:
@jreback I can't use the constructor in that case because it stacks a list of series horizontally and there's no axis option. so I used `pd.concat`. Is that okay?
```
>>> import pandas as pd
>>> s1 = pd.Series([1, 2])
>>> s2 = pd.Series([3, 4])
>>> pd.DataFrame([s1, s2])
   0  1
0  1  2
1  3  4
```